### PR TITLE
Call template methods instead of copy and paste

### PR DESCRIPTION
**Pull request by youtalk**
I found almost the same logic around `JointStateHandle`, `JointCommandHandle`, and `OperationModeHandle` methods in `robot_hardware.cpp`.
This PR calls the template methods instead of copy and paste the same logic.

Merge after:
- [ ] #99

### CODE REVIEW

**Comment by youtalk** [comment]
Does it have too much diff? I can separate it into 3 PR. Please tell me if you need.

**Comment by youtalk** [comment]
Welcome any feedback.

**Comment by destogl** [comment]
This change seams reasonable. @bmagyar what do you think?

File: hardware_interface/src/robot_hardware.cpp [hunk 1]
```diff
@@ -25,18 +25,27 @@
...
+constexpr char kJointStateLoggerName[] = "joint state handle";
+constexpr char kJointCommandLoggerName[] = "joint cmd handle";
+constexpr char kOperationModeLoggerName[] = "joint operation mode handle";
+
 template<typename T>
 hardware_interface_ret_t
-register_handle(std::vector<T *> & registered_handles, T * handle)
+register_handle(
+  std::vector<T *> & registered_handles, T * handle,
+  const std::string & logger_name)
```
Comment (Karsten1987) [inline hardware_interface/src/robot_hardware.cpp:36]: nit: the argument here could fit on one line. Either line break after each argument or no line break at all.

Comment (youtalk) [reply hardware_interface/src/robot_hardware.cpp]: Fixed. https://github.com/ros-controls/ros2_control/pull/78/commits/3e9744cac845d24d5ad42f43e20fcc3011316a93

File: hardware_interface/src/robot_hardware.cpp [hunk 2]
```diff
@@ -19,172 +19,132 @@
...
+    registered_operation_mode_handles_,
+    operation_mode_handle);
 }
 
+template<typename T>
 hardware_interface_ret_t
-RobotHardware::get_joint_state_handle(
-  const std::string & name, const JointStateHandle ** joint_state_handle)
+get_handle(
+  std::vector<T *> & registered_handles, const std::string & name, T ** handle,
```
[the remark this answers is not in the document]
Comment (youtalk) [reply hardware_interface/src/robot_hardware.cpp]: Fixed. https://github.com/ros-controls/ros2_control/pull/78/commits/3095a59a764f3b6924d4387fdb0ae53bdac216cd

**Comment by youtalk** [comment]
@Karsten1987 Thank you for your review. I've fixed them.

**Pull request merged**